### DISCUSSION TITLE
Pivot Point : Getting null value in S3,R3 and R3 R4

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#668] [Ideas]
Hi,

While using pivot indicator getting null value for S3, R3 and S4, R4,

Attaching historical data of 15 min time frame of Symbol TCS Exchange NSE and 
Attaching Pivot calculation from our GetPivotpoints() function

https://github.com/DaveSkender/Stock.Indicators/tree/main/src/m-r/PivotPoints
[TCS_PivotPoints.csv](https://github.com/DaveSkender/Stock.Indicators/files/7745772/TCS_PivotPoints.csv)
[TCS_15minuts_Historical.csv](https://github.com/DaveSkender/Stock.Indicators/files/7745773/TCS_15minuts_Historical.csv)

**Comment by DaveSkender** [comment]
See https://github.com/DaveSkender/Stock.Indicators/discussions/274#discussioncomment-1652071

**Reply by myalgomate** [reply]
KITE(ChartIQ)
--
Pivot point = (H + L + C) / 3
First support level (S1) = (2 * P) – H
Second support level (S2) = P – (H – L)
Third support level (S3) = P – 2*(H-L)
First resistance level (R1) = (2 * P) – L
Second resistance level (R2) = P + (H – L)
Third resistance level (R3) = P + 2*(H – L)

**Reply by DaveSkender** [reply]
This should be relatively easy to contribute.  Want to give it a try?

**Reply by myalgomate** [reply]
Yeah sure,
Not sure how to contribute. Will need your guidance.

**Reply by DaveSkender** [reply]
Simple explainer: 1) fork the repo, 2) create a branch in your repo, then 3) create a Pull Request back to this repo with your code.  Happy to assist along the way.  More info: https://github.com/DaveSkender/Stock.Indicators/blob/main/docs/contributing.md#developing

**Reply by myalgomate** [reply]
Thanks.
I have just tried my contribution.
Can you please verify?

**Reply by DaveSkender** [reply]
Go ahead and create a PR from within your Fork repo.  This will allow us to do a code review.  It should look like this when setting:
![image](https://user-images.githubusercontent.com/8432125/146968338-b6421098-fa3e-47a6-845d-40cb6662402c.png)

**Reply by myalgomate** [reply]
![image](https://user-images.githubusercontent.com/76814842/147036099-8f8be26f-e52c-4685-9577-db71c8199287.png)

**Reply by myalgomate** [reply]
Can you please update the library release as well.??

**Reply by DaveSkender** [reply]
This is now available in version 1.20.2
